hub/core/meta/encode/chunk_id.py

```diff
@@ -0,0 +1,235 @@
+from hub.util.exceptions import ChunkIdEncoderError
+import hub
+from hub.core.storage.cachable import Cachable
+from io import BytesIO
+from typing import Optional, Tuple
+import numpy as np
+from uuid import uuid4
+
+
+CHUNK_ID_INDEX = 0
+LAST_INDEX_INDEX = 1
+
+
+class ChunkIdEncoder(Cachable):
+    def __init__(self):
+        """Encodes chunk IDs such that they can be mapped to and from sample indices.
+
+        Note:
+            This map has a time complexity of `O(log(N))`, where `N` is the number of unique chunk IDs in this instance.
+            If this encoder is sharded, the time complexity's `N` is the max number of unique chunk IDs allowed in a single shard
+                at the worst case scenario.
+        """
+
+        self._encoded_ids = None
+        self._encoded_connectivity = None
+
+    def tobytes(self) -> memoryview:
+        bio = BytesIO()
+        np.savez(
+            bio,
+            version=hub.__encoded_version__,
+            ids=self._encoded_ids,
+            connectivity=self._encoded_connectivity,
+        )
+        return bio.getbuffer()
+
+    @staticmethod
+    def name_from_id(id: np.uint64) -> str:
+        return hex(id)[2:]
+
+    @staticmethod
+    def id_from_name(name: str) -> np.uint64:
+        return int("0x" + name, 16)
```

Comment:
Could you add this to constants and use len of this constant in name_from_id instead of 2?

Reply:
why? i don't think it lends to readability & these values aren't used anywhere else

Reply:
because the `2:` in name_from_id seems like a magic number right now

Reply:
Disagree, seems fine given the docstring